addons/project/static/src/views/burndown_chart/burndown_chart_model.js

```diff
@@ -1,14 +1,84 @@
 /** @odoo-module **/
 
 import { GraphModel } from "@web/views/graph/graph_model";
+import { sortBy } from "@web/core/utils/arrays";
 
 export class BurndownChartModel extends GraphModel {
+    /**
+     * @override
+     */
+    setup(params) {
+        super.setup(params);
+        this.stageSeqAndNamePerId = {};
+    }
+
+    /**
+     * Fetch the sequence of each stage in the project. This function alters this.stageSeqAndNamePerId
+     * @protected
+     * @param {Object} context
+     */
+    async _fetchStageInfo(context) {
+        const searchDomain =
+            !context.active_id || !context.default_project_id
+                ? []
+                : [["project_ids", "in", context.active_id]];
+        const data = await this.orm.webSearchRead("project.task.type", searchDomain, [
+            "name",
+            "sequence",
+            "id",
+        ]);
+        const stageSeqAndNamePerId = {};
+        for (const { id, name, sequence } of data.records) {
+            stageSeqAndNamePerId[id] = { name, sequence };
+        }
+        return stageSeqAndNamePerId;
+    }
+
+    /**
+     * @param {SearchParams} searchParams
+     */
+    async load(searchParams) {
+        const { context, groupBy } = searchParams;
+
+        if (groupBy.includes("stage_id")) {
+            if (context.stage_name_and_sequence_per_id && context.default_project_id) {
+                this.stageSeqAndNamePerId = context.stage_name_and_sequence_per_id;
+            } else {
+                // if the stage_name_and_sequence_per_id wasn't given by the action (for example if the page is simply reloaded)
+                this.stageSeqAndNamePerId = await this._fetchStageInfo(context);
```

Comment:
I prefer not to do an additional RPC if the infos about the stage sequence can be loaded in the action context :shrug: Do you think it's better to be consistent and always do the RPC?

Reply:
In general we avoid filling the context because if the action was defined in xml only, we cannot really do that and so we will have some inconsistency.

But here, it could be acceptable since the report is costly. 🙂